lib/datadog/tracing/tracer.rb

```diff
@@ -362,7 +362,12 @@ def bind_trace_events!(trace_op)
           event_span_op.service ||= @default_service
         end
 
+        events.propagate.subscribe do |_event_span, event_trace_op|
+          sample_trace(event_trace_op) unless event_trace_op.sampling_priority
+        end
+
         events.span_finished.subscribe do |event_span, event_trace_op|
+          sample_trace(trace_op) unless trace_op.sampling_priority
```

Comment:
`sample_trace` is now (possibly) triggered twice or more during the span lifecycle. Do we have short-circuits/guards that prevent sampling twice or help us save on performance?

Reply:
I don't see how sample_trace would be triggered twice during the span lifecycle since if it was triggered once, we'd have a sampling_priority and then we wouldn't sample. 

I'm happy to make the check internal to the method as I don't see a case where we'd ever want to run trace_sampling again after a sampling_priority value has been set.